pymc3/distributions/continuous.py

```diff
@@ -991,24 +991,53 @@ def __init__(self, distribution, lower, upper, *args, **kwargs):
         if hasattr(self.dist, 'mode'):
             self.mode = self.dist.mode
 
-    def _random(self, lower, upper, point=None, size=None):
-        samples = np.zeros(size).flatten()
+    def _random(self, lower, upper, point=None, **kwargs):
+        r"""Rejection approach to truncated sampling.
+
+        Samples from `self.dist` (i.e. the underlying distribution)
+        and accepts samples, :math:`x`, that satisfy
+        .. math::
+
+            \mathtt{lower} < x \leq \mathtt{upper}.
+
+        Parameters
+        ----------
+        lower : numpy.ndarray
+            Lower bound.  Must be comparable to `self.dist.shape`.
+        upper : numpy.ndarray
+            Upper bound.  Must be comparable to `self.dist.shape`.
+        point : dict or None
+            Conditional parameter values?
+
+        Returns
+        -------
+        numpy.ndarray
+            A single truncated sample from `self.dist`.
+
+        Notes
+        -----
+        Assumes that `self.dist` is a  `self.shape`-tensor consists of
+        i.i.d. random variables.
+        """
+        samples = np.zeros(shape=self.shape).flatten()
+        shape_len = self.dist.shape.prod()
         i, n = 0, len(samples)
         while i < len(samples):
-            sample = self.dist.random(point=point, size=n)
+            q, r = divmod(n, shape_len)
```

Comment:
Just realized that this was left over from an attempt to make `_random` behave as it used to; specifically, when it took the `size` parameter.  My initial implementation had `samples` take the shape `size + tuple(self.shape)`, then code would produce the desired result from this method alone.  However, after looking into the `generating_samples` function, I noticed that it takes steps to produce replications with shape given by `size`.